proposals/nnnn-app-services.md

```diff
@@ -0,0 +1,629 @@
+
+
+###### RPC Response Timeout
+
+The RPC response timeout should have its threshold increased when forwarding a packet to be handled by an app service provider. While processing the initial RPC request, once Core recognizes that the packet is to be forwarded the timeout for the specific request should be increased. 
+
+
+###### Unknown RPC Params
+
+Core will need to parse and store received params even if they are not listed in that version of Core's supported RPC spec. This will allow the services to be updated and function on any Core that supports the base feature. That will keep the service feature fresh and useful on older units. Essentially Core should be able to store the params received into a key/value  map and form the base RPC structs and request/responses using all entries in that map.
+
+
+##### Policy Table
+
+The IVI system can enable/disable app service providers from offering their services and consumers from using them through additions to the policy table. New functional groups will be introduced for `AppServiceProvider` and `AppServiceConsumer`. Furthermore, the functional groups can become more granular to provide access to publishing/consuming of specific app service types.
+
+
+## Potential downsides
+
+- The size of this is change is very large, and will have a wide spread effect on different aspects of all the code. This will be a complex change that will require a lot of testing.
+- RemoteControl functionality already has a subset of this feature included in the Radio module. This will cause some clashing between the two paradigms.
+
+
+## Impact on existing code
+
+Each affected platform will need to have RPCs added. Some platforms might need additional logic to be able to send notifications instead of just requests as well.
+
+### Core
+
+Core will have the biggest changes necessary. A new AppServiceManager will need to be created to manage all the published services and all the clients subscribed. The entire features should be designed to take advantage of the plug-in architecture that is used by the Remote Control feature. Core changes might require a separate proposal to ensure the architecture lines up with the SDLC's expectations. 
+
+### Mobile Libraries
+
+The mobile libraries will need a nice, developer friendly manager to handle publishing their service. These managers will need to have their own proposal as this one simply is proposing to create the functionality between the systems.
+
+### Policy Server
+
+New functional groups should be created based on [this section](#policy-table).
+
+
+
+
+## Alternatives considered
+
+- Instead of strict struct definitions, we could create very loosely defined key value pair maps. The keys would be obtained from the metadataTags enum.
+- Instead of a string to designate an image file to retrieve for service data, actually attaching the raw bytes in a param. 
```

Comment:
I think this should be updated to "...to retrieve service data, actually attach the raw bytes..."